[Bug 18714][Bug 18779] segmented: Documentation & property improvements

peter-b opened the pull request:
- Properties are now updated immediately, rather than at the next
  redraw
- Each of the list-like properties (`itemNames`, `itemLabels`, etc.)
  now always has `itemCount` items
- The `itemNames` is now permitted to have duplicated and/or empty
  elements, even in last position (bug 18714)
- All properties have been fully documentend and cross-linked with
  each other
- Dividers are no longer drawn between segments when `showBorder` is
  false, and segments are hinted so that no background shows through
  between them
- Tests, lots of tests

livecodeali commented on extensions/widgets/segmented/tests/highlight.livecodescript:
```diff
@@ -0,0 +1,165 @@
+-- Local Variables:
+-- tab-width: 3
+-- indent-tabs-mode: t
+-- End:
```
Is this stuff supposed to be here?

peter-b replied:
Um, I added that while editing the scripts in Emacs to make it nicer. I can remove them if you like. 😶

montegoulding replied:
@peter-b if you indent livecode script with tabs then every time someone touches it with the IDE editor it will cause whitespace diff headaches.

peter-b commented:
Things that need to be done:
- [x] Move Emacs variables into `.dir-locals.el`
- [x] Factor out string-list helper functions and put them into widgetutils module
- [ ] Audit other widgets that have string-list properties and get them to use helper functions

peter-b committed:
tests: Recognise LCB "test" handlers case-insensitively

Test handlers in LCB unit test no longer need to start with the exact
string `Test` -- `test` and `TEST` are also recognized.

peter-b committed:
scriptitems: Add a new extension module for manipulating LCS items

This patch adds a new "scriptitems" module extension.  It provides
utility functions for working with LiveCode Script item lists from
LCB, converting them to-and-from LCB `List` values.

peter-b committed:
[Bug 18714][Bug 18799] segmented: Documentation & property improvements

- Properties are now updated immediately, rather than at the next
  redraw
- Each of the list-like properties (`itemNames`, `itemLabels`, etc.)
  now always has `itemCount` items
- The `itemNames` is now permitted to have duplicated and/or empty
  elements, even in last position (bug 18714)
- All properties have been fully documentend and cross-linked with
  each other
- Dividers are no longer drawn between segments when `showBorder` is
  false, and segments are hinted so that no background shows through
  between them
- Tests, lots of tests

peter-b committed:
scriptitems: Add to installer manifest.

livecodeali commented:
@livecode-vulcan review ok 3048cf5

livecode-vulcan commented:
💙 review by @livecodeali ok 3048cf573ba24aa97bd9a682077c00f925f361ae

livecode-vulcan commented:
😎 test success 3048cf573ba24aa97bd9a682077c00f925f361ae
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/577)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/582)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/577)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/581)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/586)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/578)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/578)
